### PR TITLE
pythonPackages.pylint: 2.2.3 -> 2.3.1 

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
pylint was broken for python3.5 an python3.6 as mentioned in #61646 

looks like bumping the version fixed some of their unit tests that were failing with 3.5-3.6 interpreters.

`nix-build -A python36Packages.pylint `
`nix-build -A python35Packages.pylint `
`nix-build -A python37Packages.pylint `
`nix-build -A pythonPackages.pylint `

now all work.
I only tested the pytest executable wrapper of build outputs, and it seems to work as expected.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Closure for python37Packages.pylint: 102.4M -> 103.3M (the others were broken)

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build pythonPackages.pylint python3Packages.pylint

File: pkgs/development/python-modules/astroid/default.nix [hunk 1]
```diff
@@ -5,19 +5,18 @@
...
   src = fetchPypi {
     inherit pname version;
-    sha256 = "08hz675knh4294bancdapql392fmbjyimhbyrmfkz1ka7l035c1m";
+    sha256 = "1x5c8fiqa18frwwfdsw41lpqsyff3w4lxvjx9d5ccs4zfkhy2q35";
   };
 
   # From astroid/__pkginfo__.py
-  propagatedBuildInputs = [ lazy-object-proxy six wrapt ]
-    ++ lib.optional (pythonOlder "3.5") typing
-    ++ lib.optional (pythonOlder "3.7" && !isPyPy) typed-ast;
```
Comment (marsam) [inline pkgs/development/python-modules/astroid/default.nix]: typed-ast is still [restricted to cython](https://github.com/PyCQA/astroid/compare/astroid-2.1.0...astroid-2.2.5#diff-1e8a54c814195e2e0b781b5c41b2bf58R33), can you add this back?
```nix
 ++ lib.optional (!isPyPy) typed-ast
```
Besides that, LGTM

**Comment by jonringer** [comment]
@GrahamcOfBorg build pythonPackages.pylint python3Packages.pylint python36Packages.pylint python35Packages.pylint

File: pkgs/development/python-modules/astroid/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/astroid/default.nix]: Forced pushed update, should be fixed now.

**Comment by marsam** [comment]
Tested locally on NixOS and darwin
Thanks!

**Comment by vbgl** [comment]
The astroid update broke asttokens (on python 3).

**Comment by marsam** [comment]
looks like upstream fixed it https://github.com/gristlabs/asttokens/commit/21caaaa74105c410b3d84c3d8ff0dc2f612aac9a. I'm going to update it later

**Pull request merged**